mitosheet/mitosheet/parser.py

```diff
@@ -109,6 +109,38 @@ def safe_contains(
 
     return False
 
+def safe_contains_single_equals(formula: str, column_headers: List[ColumnHeader]) -> bool:
+    """
+        Check if the string contains a single equals sign that is not part of a substring:
+
+        "A = B",                    # True
+        "IF(A=B, 1, 0)",            # True
+        'A==B',                     # False
+        "CONCAT(A, 'ABC=123')",     # False
+        'A!=B',                     # False
+        'A>=B',                     # False
+        'A<=B',                     # False
+    """
+
+    # If the column headers contain an = then, we don't continue the check 
+    # because it is too hard, and rare.
+    for column_header in column_headers:
+        if isinstance(column_header, str) and '=' in column_header:
+            return False
+
+    strings_to_ignore = [
+        '!=',
+        '>=',
+        '<=',
+    ]
+
+    # Remove all occurences of string_to_ignore from the formula
+    for string_to_ignore in strings_to_ignore:
+        formula = formula.replace(string_to_ignore, '')
+
+    pattern = r"(^|[^'\"=])\=(?![\=])([^'\"=]|$)"
```

Comment:
Maybe you could add a little explanation of this regex? I'm not great with regexes but is it possible to add in some of the logic above about ignoring strings into this regex if we're already making a regex?

Reply:
Good call! I'll add some documentation here